### PR TITLE
add: teach copyFileWithTar() about symlinks and directories

### PR DESCRIPTION
**Pull request by nalind**
Teach `copyFileWithTar()` about symbolic links and directories, and use it to produce tar data to feed to `untar()` instead of special-casing them.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @vrothberg @giuseppe @ashley-cui @QiWang19 PTAL

**Comment by vrothberg** [comment]
@rh-atomic-bot r+

File: util.go [hunk 1]
```diff
@@ -120,16 +120,13 @@ func (b *Builder) copyFileWithTar(tarIDMappingOptions *IDMappingOptions, chownOp
 		}
 	}
 	return func(src, dest string) error {
+		var f *os.File
+
 		logrus.Debugf("copyFileWithTar(%s, %s)", src, dest)
-		f, err := os.Open(src)
+		fi, err := os.Lstat(src)
```
Comment (TomSweeneyRedHat) [inline util.go]: Since we're doing the system.Lstat below at line 131, is this one required?

**Comment by TomSweeneyRedHat** [comment]
One question and a homu fail, otherwise LGTM

File: util.go (hunk 1 above)
Comment (giuseppe) [reply util.go]: system.Lstat seems like a portable way to retrieve the UID/GID, unfortunately it costs another stat as it cannot be used as a `FileInfo`

**Comment by giuseppe** [comment]
@rh-atomic-bot retry

**Pull request closed without merging**